lambdas/report-user-identity/src/main/java/uk/gov/di/ipv/core/reportuseridentity/ReportUserIdentityHandler.java

```diff
@@ -213,38 +212,47 @@ private void processUsersToFindLOCAndUpdateDb(
-                    false);
-        } else {
-            boolean anyVCsMigrated = tacticalVcs.stream().anyMatch(vc -> vc.getMigrated() != null);
-            boolean allVCsMigrated = tacticalVcs.stream().allMatch(vc -> vc.getMigrated() != null);
+    private ReportUserIdentityItem evaluateUserIdentityDetail(String userId) {
+        try {
+            var tacticalVcs = verifiableCredentialService.getVcs(userId);
 
-            if (anyVCsMigrated && !allVCsMigrated) {
-                LOGGER.warn(
-                        LogHelper.buildLogMessage("Not all VCs are migrated for this user.")
+            if (!userIdentityService.areVcsCorrelated(tacticalVcs)) {
+                LOGGER.info(
+                        LogHelper.buildLogMessage("User VCs not correlated.")
                                 .with("user_hash", ReportUserIdentityItem.getUserHash(userId)));
-            }
+                return new ReportUserIdentityItem(
+                        userId,
+                        Vot.P0.name(),
+                        tacticalVcs.size(),
+                        reportUserIdentityService.getIdentityConstituent(tacticalVcs),
+                        false);
+            } else {
+                boolean anyVCsMigrated =
+                        tacticalVcs.stream().anyMatch(vc -> vc.getMigrated() != null);
+                boolean allVCsMigrated =
+                        tacticalVcs.stream().allMatch(vc -> vc.getMigrated() != null);
+
+                if (anyVCsMigrated && !allVCsMigrated) {
+                    LOGGER.warn(
+                            LogHelper.buildLogMessage("Not all VCs are migrated for this user.")
+                                    .with("user_hash", ReportUserIdentityItem.getUserHash(userId)));
+                }
 
-            return new ReportUserIdentityItem(
-                    userId,
-                    reportUserIdentityService
-                            .getStrongestAttainedVotForCredentials(tacticalVcs)
-                            .orElse(Vot.P0)
-                            .name(),
-                    tacticalVcs.size(),
-                    reportUserIdentityService.getIdentityConstituent(tacticalVcs),
-                    allVCsMigrated);
+                return new ReportUserIdentityItem(
+                        userId,
+                        reportUserIdentityService
+                                .getStrongestAttainedVotForCredentials(tacticalVcs)
+                                .orElse(Vot.P0)
+                                .name(),
+                        tacticalVcs.size(),
+                        reportUserIdentityService.getIdentityConstituent(tacticalVcs),
+                        allVCsMigrated);
+            }
+        } catch (HttpResponseExceptionWithErrorBody | CredentialParseException ex) {
+            LOGGER.warn(
+                    LogHelper.buildErrorMessage(
+                            "Exception while retrieving user VCs or vc having missing name.", ex));
```

Comment:
Can we include the hashed user id as well, otherwise this'll be very hard to diagnose...

Reply:
Done